File: CPP/HammingWeight.cpp
/*
  Hamming weight is counting the number of ones in a unsigned integer. This can be done by doing an & with 1 until the value is 0, keeping a counter.
*/
class Solution
{
public:
  int hammingWeight(int n)
  {
    unsigned int num = (unsigned int)n;
    int numOnes = 0;

    while (num != 0)
    {
      numOnes += num & 1;
      num = num >> 1;
    }

    return numOnes;
  }
};

File: JS/HammingWeight.js
/**
 * Take an unsigned integer and return the number of 1 bits it has.
 * @param {number} 
 * @return {number} number of 1's
 */
const hammingWeight = function(n) {
	let ones = 0;

	while (n != 0) {
		ones += n & 1;

		// Unsiged right shift. Arithmetic signed, logical unsigned.
		n = n >>> 1;
	}

	return ones;
};

// So basically n continues to check until 1 is nothing.
